tests/unit_tests/test_config.py

```diff
@@ -146,4 +150,91 @@ def test_auth_from_env_throws_when_not_available():
     with pytest.raises(KeyError):
         BasicAuthentication(username="${BAZ}", password="baz")
     with pytest.raises(KeyError):
-        BasicAuthentication(username="${baz}", password="baz")
+        BasicAuthentication(username="${baz}", passcode="baz")
+
+
+def is_subset(subset: Mapping[str, Any], superset: Mapping[str, Any]) -> bool:
+    """
+    Recursively check if 'subset' is contained within 'superset',
+    skipping nullable (None) fields in the superset.
+    """
+    for key, value in subset.items():
+        if key not in superset:
+            return False
+        superset_value = superset[key]
+
+        # If both values are dictionaries, recurse
+        if isinstance(value, dict) and isinstance(superset_value, dict):
+            if not is_subset(value, superset_value):
+                return False
+        # Check equality for non-dict values, ignoring None in superset
+        elif superset_value is not None and value != superset_value:
+            return False
+    return True
+
+
+# Parameterize the fixture to accept different config examples
+@pytest.fixture
+def temp_yaml_config_file(request):
+    # Use the provided config data from test parameters
+    config_data = request.param
+
+    # Create a temporary YAML file with the configuration
+    with tempfile.NamedTemporaryFile(
+        suffix=".yaml", mode="w", delete=False
+    ) as temp_yaml_file:
+        yaml.dump(config_data, temp_yaml_file)
+        temp_yaml_file_path = temp_yaml_file.name
+
+    # Provide the path and the config data
+    yield Path(temp_yaml_file_path), config_data
+
+    # Cleanup after test execution
+    os.remove(temp_yaml_file_path)
+
+
+# Parameterized test to run with different configurations
+@pytest.mark.parametrize(
+    "temp_yaml_config_file",
+    [
+        # Different configuration examples passed to the fixture
+        {
+            "env": {
+                "sources": [
+                    {"kind": "dodal", "module": "dodal.adsim"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.plans"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.stubs"},
+                ],
+            },
+            "api": {"host": "0.0.0.0", "port": 8000},
+        },
+        {
+            "stomp": None,
+            "env": {
+                "sources": [
+                    {"kind": "dodal", "module": "dodal.adsim"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.plans"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.stubs"},
+                ],
+                "events": {"broadcast_status_events": True},
+            },
+            "logging": {"level": "INFO"},
+            "api": {"host": "0.0.0.0", "port": 8000, "protocol": "http"},
+            "scratch": None,
+        },
+    ],
+    indirect=True,
+)
+def test_config_yaml_parsed(temp_yaml_config_file):
+    temp_yaml_file_path, config_data = temp_yaml_config_file
```

Comment:
Could: Split into two fixtures, one depending on the other, and inject them into the test as separate arguments. It would look neater.

Reply:
will do

Reply:
> look neater

at the price of two fixtures that depend on one another. As a `could`, I choose to skip this one